File: dehaze-android/sdk/src/main/java/com/pei/dehaze/sdk/model/prediction/DehazeParams.java
package com.pei.dehaze.sdk.model.prediction;

import lombok.Data;

/**
 * 去雾处理通用参数。
 * 范围约束对齐产品文档《去雾处理需求规格》：
 * <ul>
 *     <li>去雾强度 strength：0-100，默认 50</li>
 *     <li>色彩饱和度 saturation：0-200，默认 100（100 为原始值）</li>
 *     <li>对比度 contrast：0-200，默认 100（100 为原始值）</li>
 *     <li>锐化程度 sharpen：0-100，默认 30</li>
 * </ul>
 */
@Data
public class DehazeParams {
    /** 去雾强度 0-100 */
    private int strength;
    /** 色彩饱和度 0-200 */
    private int saturation;
    /** 对比度 0-200 */
    private int contrast;
    /** 锐化程度 0-100 */
    private int sharpen;

    public DehazeParams() {
        this(50, 100, 100, 30);
    }

    public DehazeParams(int strength, int saturation, int contrast, int sharpen) {
        this.strength = strength;
        this.saturation = saturation;
        this.contrast = contrast;
        this.sharpen = sharpen;
    }

    /**
     * 校验参数范围，返回首个不合法字段的错误提示，全部合法返回 null。
     */
    public String validate() {
        if (strength < 0 || strength > 100) {
            return "去雾强度须在 0-100 之间";
        }
        if (saturation < 0 || saturation > 200) {
            return "色彩饱和度须在 0-200 之间";
        }
        if (contrast < 0 || contrast > 200) {
            return "对比度须在 0-200 之间";
        }
        if (sharpen < 0 || sharpen > 100) {
            return "锐化程度须在 0-100 之间";
        }
        return null;
    }
}

File: dehaze_flutter/lib/models/dehaze_params.dart
/// 去雾处理通用参数
///
/// 对齐产品文档 §2.3 通用算法参数：
/// - strength 去雾强度（0-100，默认 50）
/// - saturation 饱和度（0-200，默认 100）
/// - contrast 对比度（0-200，默认 100）
/// - sharpen 锐化（0-100，默认 30）
///
/// 强类型替代 `Map<String, dynamic>`，避免字符串 key 拼写错误无法在编译期发现。
/// [toJson] 仅包含与默认值不同的字段，避免干扰算法默认行为。
class DehazeParams {
  const DehazeParams({
    this.strength = 50,
    this.saturation = 100,
    this.contrast = 100,
    this.sharpen = 30,
  });

  /// 去雾强度 0-100
  final int strength;

  /// 饱和度 0-200
  final int saturation;

  /// 对比度 0-200
  final int contrast;

  /// 锐化 0-100
  final int sharpen;

  /// 默认参数实例
  static const DehazeParams defaultParams = DehazeParams();

  /// 是否全部为默认值（此时应传 null，不干扰算法默认行为）
  bool get isDefault =>
      strength == 50 && saturation == 100 && contrast == 100 && sharpen == 30;

  /// 序列化为请求参数 Map，仅包含与默认值不同的字段
  Map<String, dynamic> toJson() {
    final result = <String, dynamic>{};
    if (strength != 50) result['strength'] = strength;
    if (saturation != 100) result['saturation'] = saturation;
    if (contrast != 100) result['contrast'] = contrast;
    if (sharpen != 30) result['sharpen'] = sharpen;
    return result;
  }
}
